### PR TITLE
Adding linter to the gate for zuul

### PR DESCRIPTION
**Pull request by matburt**

### CODE REVIEW

**Comment by matburt** [comment]
recheck

**Comment by matburt** [comment]
recheck

**Comment by matburt** [comment]
recheck

**Comment by matburt** [comment]
recheck

**Comment by matburt** [comment]
recheck

**Comment by matburt** [comment]
recheck

**Comment by matburt** [comment]
recheck

**Comment by matburt** [comment]
recheck

**Comment by matburt** [comment]
recheck

File: test/unit/test___main__.py [hunk 1]
```diff
@@ -1,234 +0,0 @@
-import os
```
Comment (benthomasson) [inline test/unit/test___main__.py]: I vote for leaving the tests in the repo.

Comment (matburt) [reply test/unit/test___main__.py]: These tests call Runner directly as part of the unit test suite... that is antithetical to what unit tests are supposed to do.

Comment (benthomasson) [reply test/unit/test___main__.py]: I prefer misclassified tests to no tests.

**Pull request merged**